internal/backends/collection_test.go

```diff
@@ -24,10 +24,203 @@ import (
+				must.NotFail(types.NewDocument("_id", types.ObjectID{1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0})),
+			}
+
+			_, err = coll.InsertAll(ctx, &backends.InsertAllParams{Docs: insertDocs})
+			require.NoError(t, err)
+
+			_, err = cappedColl.InsertAll(ctx, &backends.InsertAllParams{Docs: insertDocs})
+			require.NoError(t, err)
+
+			t.Run("CappedCollection", func(t *testing.T) {
+				t.Parallel()
+
+				queryRes, err := cappedColl.Query(ctx, new(backends.QueryParams))
+				require.NoError(t, err)
+
+				docs, err := iterator.ConsumeValues[struct{}, *types.Document](queryRes.Iter)
+				require.NoError(t, err)
+				require.Len(t, docs, len(insertDocs))
+
+				for i, doc := range docs {
+					// inserted doc is frozen, queried doc is not so compare each value
+					assert.NotZero(t, doc.RecordID())
+					assert.Equal(t, insertDocs[i].RecordID(), doc.RecordID())
+					assert.Equal(t, insertDocs[i].Keys(), doc.Keys())
+					assert.Equal(t, insertDocs[i].Values(), doc.Values())
+				}
+
+				explainRes, err := cappedColl.Explain(ctx, new(backends.ExplainParams))
+				require.NoError(t, err)
+				assert.True(t, explainRes.SortPushdown)
+			})
+
+			t.Run("CappedCollectionOnlyRecordIDs", func(t *testing.T) {
+				t.Parallel()
+
+				queryRes, err := cappedColl.Query(ctx, &backends.QueryParams{OnlyRecordIDs: true})
+				require.NoError(t, err)
+
+				docs, err := iterator.ConsumeValues[struct{}, *types.Document](queryRes.Iter)
+				require.NoError(t, err)
+				require.Len(t, docs, len(insertDocs))
+
+				// inserted doc is frozen, queried doc is not frozen hence compare each value
+				for i, doc := range docs {
+					assert.NotZero(t, docs[i].RecordID())
+					assert.Equal(t, insertDocs[i].RecordID(), doc.RecordID())
+					assert.Empty(t, docs[i].Values())
+				}
+
+				explainRes, err := cappedColl.Explain(ctx, new(backends.ExplainParams))
+				require.NoError(t, err)
+				assert.True(t, explainRes.SortPushdown)
+			})
+
+			t.Run("CappedCollectionSortAsc", func(t *testing.T) {
+				if name == "sqlite" {
+					t.Skip("https://github.com/FerretDB/FerretDB/issues/3181")
+				}
+
+				t.Parallel()
+
+				sort := backends.SortField{Key: "_id"}
+				queryRes, err := cappedColl.Query(ctx, &backends.QueryParams{Sort: &sort})
+				require.NoError(t, err)
+
+				docs, err := iterator.ConsumeValues[struct{}, *types.Document](queryRes.Iter)
+				require.NoError(t, err)
+				require.Len(t, docs, len(insertDocs))
+
+				// inserted doc is frozen, queried doc is not frozen hence compare each value
+				assert.Equal(t, insertDocs[2].RecordID(), docs[0].RecordID())
+				assert.Equal(t, insertDocs[2].Values(), docs[0].Values())
+
+				assert.Equal(t, insertDocs[0].RecordID(), docs[1].RecordID())
+				assert.Equal(t, insertDocs[0].Values(), docs[1].Values())
+
+				assert.Equal(t, insertDocs[1].RecordID(), docs[2].RecordID())
+				assert.Equal(t, insertDocs[1].Values(), docs[2].Values())
```

Comment:
In general: if you find yourself doing that over and over again, search for existing helper or add one :)

Reply:
:+1: